apollo-at-link/src/spec_definition.rs

```diff
@@ -0,0 +1,157 @@
+use crate::federation_schema::{FederationSchemaRef, OptionLinksMetadata};
+use crate::link::Link;
+use crate::spec::{Identity, Url, Version};
+use apollo_compiler::schema::{DirectiveDefinition, ExtendedType};
+use apollo_compiler::{Node, NodeStr};
+use std::collections::btree_map::Keys;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+pub trait SpecDefinition {
```

Comment:
What's the reasoning for this to be a trait rather than just a struct? From reading, it seems like it could just be a struct and that would help remove some of the generics that get carried along some parts of the code.